source/includes/api/_query.md

```diff
@@ -1,204 +1,278 @@
-SELECT * FROM table limit=20 offset 10
+```shell
+curl -X GET 'https://api.resourcewatch.org/v1/download/<dataset.id>?sql=SELECT * from <dataset.tableName>'
 ```
 
+> Requesting the download of a query providing the dataset ID in the FROM clause of the query:
 
-## Raster queries available
-
-```sql
-SELECT ST_METADATA(rast) FROM table
-SELECT ST_BANDMETADATA(rast, occurrence) FROM table
-SELECT ST_SUMMARYSTATS() FROM table
-SELECT ST_HISTOGRAM(rast, 1, auto, true) FROM table
-SELECT ST_valueCount(rast, 1, true) FROM table
+```shell
+curl -X GET 'https://api.resourcewatch.org/v1/download?sql=SELECT * from <dataset.id>'
 ```
 
-## Freeze query
-
-It is possible generate a json file in a bucket of the sql result. You only need send a query param freeze with value true and you will obtain the url of the json file.
-
-<aside class="notice">
-    This is an authenticated feature!
-</aside>
+> Explicitly setting the format of the returned file:
 
 ```shell
-curl -i -XGET http\://api.resourcewatch.org/v1/query\?sql\=SELECT\ \*\ from\ <dataset.slug>&freeze=true
+curl -X GET 'https://api.resourcewatch.org/v1/download?sql=SELECT * from <dataset.id>&format=json'
+curl -X GET 'https://api.resourcewatch.org/v1/download?sql=SELECT * from <dataset.id>&format=csv'
 ```
 
-## Rasdaman queries
+You can download the result of a query using the `download` endpoint. You can either provide the dataset id in the URL or you can pass it as part of the SQL query (in the FROM clause).
+
+You can also specify which file type you want to download (JSON or CSV) - except for Google Earth Engine datasets (which only supports JSON). By default, the API will return a CSV file (or a JSON file for Google Earth Engine).
+
+**Please not that not all dataset providers support downloading queries - the following providers support downloading query results:**
+
+* Google Earth Engine
+* Document
+* Carto
+* BigQuery
+* ArcGIS FeatureService
+
+The response body will contain the data to be downloaded. In the case of the format `json`, the returned result will be a JSON object with the results of the execution of the query provided. In the case of format `csv`, the body of the response will contain the actual CSV data corresponding to the results of the execution of the query provided.
+
+Calling the download endpoint might sometimes result in an error being returned. The following table describes the possible errors that can occur when downloading query results:
+
+Error code     | Error message  | Description
+-------------- | -------------- | --------------
+400            | SQL o FS required | The required `sql` field is missing either as query string parameter or in the request body.
+500            | Internal server error | The error message might vary in this case.
+
+## Supported SQL syntax reference
+
+### CartoDB datasets
+
+This section describes the SQL support for querying datasets with provider `cartodb`.
+
+| Supported | Feature | Example URL |
```

Comment:
Could use more visual demarcation between yes and no rows - maybe shade the no rows (does Markdown support that?) or separate into two tables?

Reply:
Hum, highlighting rows would be the best option here in my opinion, but as far as I know, that is not possible with Markdown... 

I'd prefer keeping only one table because queries are grouped logically by the clause they refer to - if we had two tables, users would need to check both tables to see which methods are allowed when selecting columns for instance...

Would strong emphasis with bold work for this (example below :point_down:)?
![Screenshot from 2020-05-12 11-27-32](https://user-images.githubusercontent.com/2981676/81673480-9e6ec280-9443-11ea-8d57-f6758eb3e10a.png)

Reply:
Sure, that's good for now. Since it's just a stylistic thing don't spend too much time on it. We can revisit at a later time. At some point towards the end of the contract I'll probably go through and make things a little prettier in general.

Reply:
Done :+1: